app/drivers/sensor/ec11/ec11.h

```diff
@@ -15,6 +15,7 @@ struct ec11_config {
     const struct gpio_dt_spec b;
 
     const uint8_t resolution;
+    const uint8_t steps;
```

Comment:
I haven't seen such an encoder yet, but it seems like it should be possible for an encoder to exists that has more than 255 steps.

Reply:
Yes, the upcoming ma730 hall effect encoder has super high resolution. But ec11 encoders with that high a resolution... Haven't seen any *yet*... I suppose they could exist though.

Reply:
By way of DigiKey search, I found one optical encoder with quadrature output at 512 pulses per revolution, and two magnetic encoders with quadrature output and 256 pulses per revolution. Seems unlikely that they'd be used, but you did make triggers_per_rotation be a uint16, so it seems reasonable to me for steps to also be one.

Reply:
Yeah, fair. Bumped it to 16-bit.